prov/tcp/src/tcpx_ep.c

```diff
@@ -199,13 +199,13 @@ static int tcpx_bind_to_port_range(SOCKET sock, void* src_addr, size_t addrlen)
 		ofi_addr_set_port(src_addr, rand_port_number);
 		ret = bind(sock, src_addr, (socklen_t) addrlen);
 		if (ret) {
-			if (errno == EADDRINUSE)
+			if (ofi_sockerr() == EADDRINUSE)
 				continue;
 
 			FI_WARN(&tcpx_prov, FI_LOG_EP_CTRL,
 				"failed to bind listener: %s\n",
 				strerror(ofi_sockerr()));
-			return -errno;
```

Comment:
I would store errno, into another var as FI_WARN() has the potential risk to change errno. Hmm, the other returns above have the same issue.

Reply:
Have you see errno change with FI_WARN?  I didn't think the fi_log path calls anything that sets errno.

Reply:
I didn't get it with FI_WARN() yet, but I got overwritten errnos in my long distant past from printf functions. It is not mentioned in the linux man page, but at least opengroup states

<quote>
If an output error was encountered, these functions shall return a negative value [CX] [Option Start]  and set errno to indicate the error. 
</quote>

https://pubs.opengroup.org/onlinepubs/9699919799/functions/fprintf.html

Reply:
Hmm... I think the safest option here would be to modify the low-level fi_log() call to save and restore errno.  My guess is there are dozens of places in the code where errno is returned after calling one of the FI_LOG functions.  This shouldn't impact performance.

Reply:
Ok fine, going to create a patch for that.

Reply:
Let me know if you want me to do the patch

Reply:
Let me know me if you want me to do the patch. Thanks!